tests/test-lexer.cpp

```diff
@@ -65,9 +65,9 @@ auto test_regex_ast(string_view var_schema, u32string const& expected_serialized
  * The lexer's symbol mappings are initialized based on the schema variables.
  * Assumes the lexer is in a clean state before initialization.
  * @param schema_ast Contains the variables to add to the lexer.
- * @param lexer Returns the initialized parser.
+ * @return The initialized lexer.
  */
-auto initialize_lexer(std::unique_ptr<SchemaAST> schema_ast, ByteLexer& lexer) -> void;
+auto initialize_lexer(std::unique_ptr<SchemaAST> schema_ast) -> ByteLexer;
```

Comment:
```suggestion
[[nodiscard]] auto create_lexer(std::unique_ptr<SchemaAST> schema_ast) -> ByteLexer;
```
How about rename to `create_lexer`?

Reply:
Changed.